components/dashboard/donut.tsx

```diff
@@ -0,0 +1,86 @@
+import { Chart as ChartJS, ArcElement, Legend, Tooltip } from "chart.js";
+import { Doughnut } from "react-chartjs-2";
+import { defi } from "../../utils/tzktHooks";
+
+ChartJS.register(ArcElement, Tooltip, Legend);
+
+type DonutProperty = {
+  title: string;
+  colors: string[];
+  tokens: defi[];
+};
+
+const Donut = ({ title, colors, tokens }: DonutProperty) => {
+  // We only need 4 tokens, the 5th one is the sum of the remaining
+  const tokenSample: Array<defi> = tokens.reduceRight<Array<defi>>(
```

Comment:
Peut etre utiliser `useCallback` pour éviter les recalculs inutiles, tu veux juste recalculer quand `tokens` change

Reply:
fixed in: https://github.com/marigold-dev/tzsafe-ui/pull/167/commits/60f7cff4040219dce9d2c33309a2b3908e80a38a